### PR TITLE
Allow for overriding RPATH with higher priority paths

### PR DESCRIPTION
**Pull request by ocaisa**
This addresses https://github.com/EESSI/software-layer/issues/88 where EESSI wants the ability to override what is linked in rpath (for example to inject an ABI compatible MPI).

### CODE REVIEW

File: easybuild/framework/easyblock.py [hunk 1]
```diff
@@ -2144,6 +2144,24 @@ def prepare_step(self, start_dir=True, load_tc_deps_modules=True):
...
+                _log.debug("Converted RPATH override directories ('%s') to a list of paths: %s" % (rpath_overrides,
+                                                                                                   rpath_override_dirs))
+                for path in rpath_override_dirs:
+                    if not os.path.isabs(path):
+                        raise EasyBuildError(
+                            "Path used in rpath_override_dirs is not an absolute path: %s", path)
+            else:
+                raise EasyBuildError("Value for rpath_override_dirs has invalid type (%s), should be string: %s",
+                                     type(rpath_overrides), rpath_overrides)
+            self.rpath_include_dirs.extend(rpath_override_dirs)
```
Comment (boegel) [inline easybuild/framework/easyblock.py:2163]: @ocaisa Don't we want to add these paths at the very front, even before `<installdir>/lib` and `$ORIGIN` & co?

With the `/opt/eessi/lib` use case in mind, we want to *always* pick up libraries from there, even for binaries that come with that library in the corresponding installation directory?

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/framework/easyblock.py (hunk 1 above)
Comment (ocaisa) [reply easybuild/framework/easyblock.py:2163]: Yeah, I struggled with that and went the other way, assuming that if you are inside the installation then you want to stay consistent...but I don't know, you could argue either way

Comment (boegel) [reply easybuild/framework/easyblock.py:2163]: Let's take an MPICH library as an example...

If the MPICH installation comes with binaries that link to `libmpi`, do you want to stick with the `libmpi.so` that comes with the MPICH installation, or would you want to use the alternate `libmpi.so` you provide in (for example) `/opt/eessi/lib`?

I'd say it's the latter.

It's certainly an edge case though...

Comment (ocaisa) [reply easybuild/framework/easyblock.py:2154]: Ok, I reversed it

**Review by boegel — APPROVED** [review]

**Pull request merged**